src/numbers.lisp

```diff
@@ -203,6 +203,13 @@
         (%ash-right x y)
         (%ash-left x  y))))
 
+;;; log
+(defun log (number &optional (base nil base-p))
+  "Return the logarithm of NUMBER in the base BASE, which defaults to e."
+  (if base-p
+      (/ (log number) (log base))
+      (#j:Math:log number)))
```

Comment:
This wont throw if we call it with something other than a number, right?

Can we add a type assertion with `numberp`

Reply:
`numberp` added.

But I have some other questions about numbers in JSCL, how do we deal with:

- NaN
- Infinity
- -Infinity

as in:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/log

```javascript
Math.log(-1); // NaN
Math.log(-0); // -Infinity
Math.log(0); // -Infinity
Math.log(1); // 0
Math.log(10); // 2.302585092994046
Math.log(Infinity); // Infinity
```

Reply:
So far we hust accepted the limitation.

There were a few initiatives to at least add bigint to differentiate between integers and floats,  but we didn't polish it enough.